### ISSUE TITLE
(Feature request) positive feedback for code submission

### ISSUE THREAD

**Issue by ashwathirao** [exercism/exercism#6468]
Currently exercism using static analyser to give pre-defined feedback, does exercism have any future scope of providing positive feedback if the user had overcome a feedback  message in their code submission

**Comment by iHiD** [comment]
Yes, we will be giving positive feedback for Concept Exercises if someone reaches the optimal solution. For Practice Exercises, that will come down to each Track to decide. Giving positive feedback is much harder than giving a single piece of negative feedback, as giving negative feedback means looking for one potential issue whereas giving positive feedback involves being sure the whole solution is good.

**Comment by joshgoebel** [comment]
I think the poster was asking a very specific question about one scenario. 

Scenario:

- I submit a solution and receive automated negative feedback
- I fix my solution and resubmit
- automation finds no problems now


Theoretically the system could consider that progression itself as an improvement and pat me on the shoulder for fixing the issue.

**Comment by iHiD** [comment]
Yeah, by my (potentially badly made point) Is that there's no guarantee that replying to the feedback hasn't made the situation worse. And unless the analyzer is checking for everything, there's a risk we say "That's better" when it's actually much worse.

**Comment by joshgoebel** [comment]
> no guarantee that replying to the feedback hasn't made the situation worse

Ah, I suppose this is true./. It's only true that they did *something* to make the analyzer happy - they potentially could have done something *really bad* to make the analyzer happy- just something the analyzer isn't programmed to detect yet.  One could imagine a future where this sort of "kudos" was possible, but the analyzer would need to be MUCH smarter (both in detecting good and bad code).

@ashwathirao I think perhaps it's a reasonable idea, just we don't have the tech right now to do it without potentially misleading students - which would do more harm than good.
